src/common/Language.ts

```diff
@@ -59,7 +61,9 @@ export function defaultDetokenizer(
   whitespace: WhitespaceInsertLogic = defaultWhitespaceInsertLogic,
   indent = 1
 ): Detokenizer {
-  return function (tokens: string[]): string {
+  return function (tokenTree: TokenTree): string {
+    // @ts-expect-error
+    const tokens: string[] = [tokenTree].flat(Infinity); // it seems ts doesn't understand flattening the tree
```

Comment:
TS could be convinced to like this by doing some wrapper function + recursion, but this solution is elegant enough that it warrants the `@ts-expect-error`